app/models/bearer_token.rb

```diff
@@ -0,0 +1,25 @@
+class BearerToken
+  SOURCE_URL = "https://login.microsoftonline.com/fad277c9-c60a-4da1-b5f3-b3b8b34a82f9/oauth2/v2.0/token"
+
+  def self.token
+    new.token
+  end
+
+  def token
+    json[:access_token]
+  end
+
+  def json
+    @json ||= JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def response
+    @response ||= Faraday.post(
+      SOURCE_URL,
+      grant_type: 'client_credentials',
+      scope: "api://6db0ed68-bd2e-455f-9a27-a24ed70dd6f8/.default",
```

Comment:
Should this be set somewhere rather than hardcoded?

Reply:
I've moved the scope to the credentials.